packages/use-optionally-controlled-state/src/index.tsx

```diff
@@ -37,12 +45,17 @@ export default function useOptionallyControlledState<Value>({
     }
   }
 
-  const value = isControlled ? controlledValue : stateValue;
+  const value = isControlled ? controlledValue : (stateValue as Value);
 
-  function onValueChange(nextValue: Value) {
-    if (!isControlled) setStateValue(nextValue);
-    if (onChange) onChange(nextValue);
-  }
+  const isControlledRef = useRef(false);
+  isControlledRef.current = isControlled;
```

Comment:
I think this is not concurrent mode safe. Can we make `isControlled` a dependency of the callback? If `isControlled` changes, we throw an error anyway.

In the future we might want to use [`useEvent`](https://github.com/reactjs/rfcs/pull/220) here.

Reply:
hmm you are right. I wanted it to behave like useState (which provides a stable setState), but I didn't think about concurrency. Let me read into it

Reply:
oh no, this is not safe at all. and I used it all over the place 😱 

so that's why `useLatest` used `useEffect` to update the current value. 🤔 
oh well. I will add it to the dep array